src/structs/vpx_bool_reader.rs

```diff
@@ -144,36 +165,25 @@ impl<R: Read> VPXBoolReader<R> {
 
         let probability = branch.get_probability() as u32;
 
-        let split = 1 + (((tmp_range - 1) * probability) >> BITS_IN_BYTE);
-        let big_split = (split as u64) << BITS_IN_LONG_MINUS_LAST_BYTE;
-        let bit = tmp_value >= big_split;
+        let split = ((((tmp_range - (1 << BITS_IN_VALUE_MINUS_LAST_BYTE)) >> 8) * probability)
+            & (0xFF << BITS_IN_VALUE_MINUS_LAST_BYTE))
+            + (1 << BITS_IN_VALUE_MINUS_LAST_BYTE);
 
-        let shift;
+        assert!(tmp_range - split > 0);
```

Comment:
Only should do this assert if lzcnt is not enabled (as per previous assert). Lzcnt doesn't require a check for zero so the assert is not needed.

Reply:
Done